engine/src/exec-interface-field-chunk.cpp

```diff
@@ -2438,6 +2438,10 @@ void MCField::GetEffectiveTextFontOfCharChunk(MCExecContext& ctxt, uint32_t p_pa
 
     if (r_mixed)
         return;
+    
+    if(!MCStringIsEmpty(*t_value))
```

Comment:
Why is the check for the non-empty string necessary?

Reply:
I think the effective textFont of a *chunk* cannot be empty. It `*t_value` is nil or `kMCEmptyString` then the effective textFont of a chunk should be the effective textFont of the field (i.e. `GetEffectiveTextFont(ctxt, r_value);` from above)

Reply:
Okay so - I don't think t_value can be nil :)

The problem here is an incorrect use of the GetCharPropCharChunk method I think... It is being called in 'effective'; mode - so the method should be:
```
MCAutoStringRef t_default;
GetEffectiveTextFont(ctxt, &t_default);

GetCharPropOfCharChunk< PodFieldPropType<MCStringRef> >(ctxt, this, p_part_id, si, ei, &MCBlock::GetTextFont, true, *t_default, r_mixed, r_value);
```

Reply:
I have tried this ^^^^ with the sample stack attached to the bug report and for the first line (where its textFont is not set - it is just the textfont of the field that is set) it returns empty for the effective textFont (and r_value is NULL in this case):

<img width="1440" alt="Screenshot 2019-10-14 at 12 31 31" src="https://user-images.githubusercontent.com/5111835/66741621-997e6c00-ee7e-11e9-892b-1d40e1e2e60e.png">